### PR TITLE
Add new script to report accelerator key of currently focused object

### PR DESCRIPTION
**Pull request by codeofdusk**
### Link to issue number:
None (reported privately by @DHowett via @carlos-zamora).

### Summary of the issue:
There is no way to simply obtain the accelerator key of the currently focused object without reading its entire description.

### Description of how this pull request fixes the issue:
Adds a new script to report the accelerator key. By default, it is bound to <kbd>Shift</kbd>+<kbd>numpad 2</kbd> (desktop) and <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>NVDA</kbd>+<kbd>.</kbd> (laptop). As a mnemonic, accelerator keys are (or at least used to be) visually displayed by an underlined character.

### Testing strategy:
Tested Windows 11 Notepad (UIA), NVDA menus (MSAA), Google Chrome (MSAA), and Foobar 2000 (MSAA).

### Known issues with pull request:
None known.

### Change log entries:
== New Features ==
- Added a new command to check the keyboard shortcut of the current focus (by default, Shift+Numpad 2 on desktop and Ctrl+Shift+NVDA+. on laptop).

### Code Review Checklist:

- [x] Pull Request description:
  - description is up to date
  - change log entries
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] API is compatible with existing add-ons.
- [x] Documentation:
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English

### CODE REVIEW

File: source/globalCommands.py [hunk 1]
```diff
@@ -2177,6 +2177,24 @@ def script_reportStatusLine(self, gesture):
...
 
+	@script(
+		description=_(
+			# Translators: Description for a keyboard command which reports the
+			# accelerator key of the currently focused object.
+			"Reports the shortcut key of the currently focused object."
+		),
+		category=SCRCAT_FOCUS,
+	)
+	def script_reportFocusObjectAccelerator(self, gesture):
```
Comment (seanbudd) [inline source/globalCommands.py:2188]:
```suggestion
	def script_reportFocusObjectAccelerator(self, gesture: "inputCore.InputGesture") -> None:
```

File: source/globalCommands.py [hunk 2]
```diff
@@ -2177,6 +2177,24 @@ def script_reportStatusLine(self, gesture):
...
 
+	@script(
+		description=_(
+			# Translators: Description for a keyboard command which reports the
+			# accelerator key of the currently focused object.
+			"Reports the shortcut key of the currently focused object."
+		),
+		category=SCRCAT_FOCUS,
+	)
+	def script_reportFocusObjectAccelerator(self, gesture: "inputCore.InputGesture") -> None:
```
Comment (seanbudd) [inline source/globalCommands.py:2188]: my mistake - this does not need to be quoted

```suggestion
	def script_reportFocusObjectAccelerator(self, gesture: inputCore.InputGesture) -> None:
```

**Review by seanbudd — APPROVED** [review]

**Comment by codeofdusk** [comment]
@seanbudd Before we merge, any idea why this doesn't work in MSAA pull-down menus?

**Comment by seanbudd** [comment]
@codeofdusk I'm not sure - it would be good to get an answer on this even if it's a separate issue / discussion / PR

**Comment by codeofdusk** [comment]
@seanbudd It appears the issues I was having with MSAA apps were caused by me binding the script to a command that caused the menus to close. Setting a default gesture (based on the "current character" commands) seems to resolve the issue.

**Comment by codeofdusk** [comment]
PR now ready for review/merge.

File: user_docs/en/userGuide.t2t [hunk 3]
```diff
@@ -316,6 +316,7 @@ There are some key commands that are useful when moving with the System focus:
 | Report title | NVDA+t | NVDA+t | Reports the title of the currently active window. Pressing twice will spell the information. Pressing three times will copy it to the clipboard |
 | Read active window | NVDA+b | NVDA+b | reads all the controls in the currently active window (useful for dialogs) |
 | Report Status Bar | NVDA+end | NVDA+shift+end | Reports the Status Bar if NVDA finds one. Pressing twice will spell the information. Pressing three times will copy it to the clipboard |
+| Report Shortcut Key | shift+numpad 2 | NVDA+control+shift+. | Reports the shortcut (accelerator) key of the currently focused object. |
```
Comment (seanbudd) [inline user_docs/en/userGuide.t2t:319]: We should try to introduce key commands consistently like so

```suggestion
| Report Shortcut Key | ``shift+numpad2`` | ``NVDA+control+shift+.`` | Reports the shortcut (accelerator) key of the currently focused object. |
```

**Comment by CyrilleB79** [comment]
I am a bit concerned by the use of Numpad for the default gesture.
Numpad key are usually used for interactions with the review cursor. Shouldn't we reserve the use of these keys for the review cursor?
@Qchristensen , @lukaszgo1 what do you think?

Could we find another key combination or keep it unbound by default?

**Comment by codeofdusk** [comment]
> I am a bit concerned by the use of Numpad for the default gesture.
Numpad key are usually used for interactions with the review cursor. Shouldn't we reserve the use of these keys for the review cursor?

@CyrilleB79 Similarity to review cursor keys is *intended*:

> As a mnemonic, accelerator keys are (or at least used to be) visually displayed by an underlined character.

So this is the "current character" command with modifiers.

**Comment by CyrilleB79** [comment]
> > I am a bit concerned by the use of Numpad for the default gesture.
> > Numpad key are usually used for interactions with the review cursor. Shouldn't we reserve the use of these keys for the review cursor?
> 
> @CyrilleB79 Similarity to review cursor keys is _intended_:
> 
> > As a mnemonic, accelerator keys are (or at least used to be) visually displayed by an underlined character.
> 
> So this is the "current character" command with modifiers.

I have understood that you want to find a similarity with a character command since accelerator key is an underlined character.
But IMO, it's more interesting to separate review command from the other ones than than finding an easy-to-remember gesture for this specific comand. Indeed, the difference between review cursor, browse mode cursor and system cursor is sometimes hard to understand for some users (not me); and all what can help to make this difference more noticeable is a plus.

Anyway, I have expressed my opinion now. But if I am the only one to think this way, just go forward with this PR as is; that's not so important.

**Comment by codeofdusk** [comment]
@seanbudd Failure seems unrelated.

**Review by Qchristensen — APPROVED** [review]
Great work, will be well appreciated.

**Pull request merged**